ios/Classes/FlutterLocalNotificationsPlugin.m

```diff
@@ -141,14 +146,11 @@ - (void)initialize:(FlutterMethodCall * _Nonnull)call result:(FlutterResult _Non
     if (arguments[REQUEST_BADGE_PERMISSION] != [NSNull null]) {
         requestedBadgePermission = [arguments[REQUEST_BADGE_PERMISSION] boolValue];
     }
-    /*if (call.arguments[ON_NOTIFICATION_CALLBACK_DISPATCHER] != [NSNull null]) {
-        [self startHeadlessService:[call.arguments[CALLBACK_DISPATCHER] longValue]];
-        [self setCallbackDispatcherHandle:[call.arguments[ON_NOTIFICATION_CALLBACK_DISPATCHER] longValue] key:ON_NOTIFICATION_CALLBACK_DISPATCHER];
-    } else {
-        [persistentState removeObjectForKey:ON_NOTIFICATION_CALLBACK_DISPATCHER];
-    }*/
+    
     if(@available(iOS 10.0, *)) {
         UNUserNotificationCenter *center = [UNUserNotificationCenter currentNotificationCenter];
+        center.delegate = self;
```

Comment:
FYI i'm reverting this change unless you have any other suggestions as it looks like it results in late registration that has led to #295 where the plugin can't handle where the notification has launched the app